src/components/attendee/ProfileOverlay/ProfileOverlay.tsx

```diff
@@ -55,13 +54,26 @@ export const ProfileOverlay: React.FC<ProfileOverlayProps> = ({ profile }) => {
     getValues,
     setValue,
     control,
+    reset,
   } = useForm<UserProfileModalFormData>({
     mode: "onBlur",
     reValidateMode: "onChange",
     defaultValues,
     shouldUnregister: true,
   });
 
+  const values = getValues();
+
+  useEffect(() => {
+    // needed to reset form values if default values served for the first time are incorrect
+    if (
+      defaultValues.partyName !== values?.partyName &&
+      values?.partyName === ""
```

Comment:
[Q] But what does this check mean? Different and empty string? Might be good to add a line or two more to this comment about the condition itself.